testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java

```diff
@@ -1836,6 +1836,27 @@ public void testDoubleUpdateAndThenReadFromHive()
         });
     }
 
+    @Test(groups = HIVE_TRANSACTIONAL)
+    public void testDeleteWithNoPredicateOnTableCreatedWithCtas()
+    {
+        try {
+            onTrino().executeQuery("SET SESSION scale_writers = true");
+            onTrino().executeQuery("SET SESSION writer_min_size = '4kB'");
+            onTrino().executeQuery("SET SESSION task_writer_count = 2");
+            withTemporaryTable("test_delete_with_no_predicate_on_ctas", true, false, NONE, tableName -> {
+                onTrino().executeQuery(
+                        "CREATE TABLE test_delete_with_no_predicate_on_ctas with (transactional = true) as select * from tpch.sf1000.orders limit 100000");
+                onTrino().executeQuery("DELETE FROM test_delete_with_no_predicate_on_ctas");
```

Comment:
the triggering condition for the bug was

- data as original files
- 2+ files

so
- verify the test reproduces the error conditions
  - check `count(DISTINCT "$path")` is >= 2
  - check "$path" are directly under table location, not nested in directories
    - see https://github.com/trinodb/trino/blob/cd2787b51dff02cdde4cba8427d99e97763478f2/testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/util/TableLocationUtils.java#L44 for inspiration
-  rename the test to `testDeleteWithOriginalFiles`

Reply:
great comment, thanks